### DISCUSSION TITLE
Agressive Caching

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [serwist/serwist#59] [General]
Sometimes I think serwist cached to agressively in the dev environment (I only had this error in the dev environment). Where the my code was out of date with the code that was send to the browser.
![image](https://github.com/serwist/serwist/assets/77451351/af3228e5-2fc6-4f2a-8a83-c5871033556b)
![image](https://github.com/serwist/serwist/assets/77451351/4afa603a-1eb5-49af-8c74-6d2915867b68)
Notice that I changed the word `heart` to `hearted` as an example.

I was able to fix this problem by deleting the cache storage
![image](https://github.com/serwist/serwist/assets/77451351/c8c698dc-a54f-4e5d-a636-405a101d8410)
But this is not an long-term solution

**Comment by FleetAdmiralJakob** [comment]
NOTE: Idk if this is really a serwist problem, so it could be also a problem of Next.js but if others also experience that with serwist or a maintainer can say that serwist saves stuff into the cache storage I think this is a problem of serwist.

**Comment by DuCanhGH** [comment]
@FleetAdmiralJakob will investigate. Seems like `runtimeCaching` made this caching a bit too sticky.

**Reply by FleetAdmiralJakob** [reply]
Hi, just created a new project with serwist and got the same problem again. If I go for example from localhost to 127.0.0.1 everything works fine, because for this url serwist did not cache the hell out of it. Also seems like that this issue is related to the app router, but I'm not sure with this.

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob isn't it that service workers are not allowed to run over HTTP for non-localhost domains?

Still, that does seem to be a Serwist issue. Would you mind temporarily using Serwist preview? The preview `installSerwist` disables both precaching and runtime caching in development mode by default, unlike the stable one, which only disables precaching in that mode. This means that there should be no caching going on in development, ensuring that your code is always up to date.

**Reply by FleetAdmiralJakob** [reply]
Serwist is running on both domains 127.0.0.1 and localhost if you mean that

**Reply by FleetAdmiralJakob** [reply]
@DuCanhGH I can confirm that with the disabled caching in the 9.0.0-preview.14 version I no longer have problem like `TypeError: Cannot read properties of undefined (reading 'call')`

Project for reproduction and testing: https://github.com/FleetAdmiralJakob/chat-io

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob why are you sending a reproduction *after* your issue has been resolved? 💀

If nothing else, I'm closing this discussion :)
